packages/core/schematics/migrations/control-flow-migration/util.ts

```diff
@@ -468,9 +468,12 @@ export function processNgTemplates(
     const templates = getTemplates(template);
 
     // swap placeholders and remove
-    for (const [name, t] of templates) {
-      const replaceRegex = new RegExp(getPlaceholder(name.slice(1)), 'g');
-      const forRegex = new RegExp(getPlaceholder(name.slice(1), PlaceholderKind.Alternate), 'g');
+    for (const [nameWithHash, t] of templates) {
+      const replaceRegex = new RegExp(getPlaceholder(nameWithHash.slice(1)), 'g');
```

Comment:
nit:
```suggestion
      const name = nameWithHash.slice(1);
      const replaceRegex = new RegExp(getPlaceholder(name), 'g');
```
Since we're doing this operation 3 times